tests/cpp/test_multidevice_sharding.cpp

```diff
@@ -40,6 +40,30 @@ TEST_F(NVFuserTest, TestIsSharded) {
   EXPECT_ANY_THROW(isSharded(c));
 }
 
+TEST_F(NVFuserTest, TestPropagateSharding) {
+  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* a = makeSymbolicTensor(3);
+  TensorView* b = makeSymbolicTensor(3);
+  TensorView* c = add(a, b);
```

Comment:
just for my own understanding: does this `add` causes the automatic insertion of (one or two?) castOps? proably some bcast too right?

Reply:
This add doesn't, but I purposefully didn't apply any sharding on C for this test.
If a was a half data type, and c float then an CastOp casting a to float would be automatically inserted.